src/scrape_up/steam/steamScraper.py

```diff
@@ -0,0 +1,146 @@
+    | `csv (int)`     | Flag indicating whether to save data to CSV (1 for True, 0 for False)  |
+
+    Note: Use `help(SteamStoreScraper.Scraper)` for more information on tags.
+    """
+
+    def __init__(self):
+        self.base_url = 'https://store.steampowered.com/search/?'
+        self.cols = ['Name', 'Date', 'Platforms', 'Original Price', 'Discount %', 'Discount Price', 'Reviews', 'Review Count', 'Filter']
+
+    def _get_total_pages(self, url):
+        """
+        Retrieves the total number of pages for a given URL.
+
+        Args:
+            url (str): The URL to scrape.
+
+        Returns:
+            int: Total number of pages.
+        """
+        response = requests.get(url)
+        doc = BeautifulSoup(response.content, 'html.parser')
+        total_pages = int(doc.find('div', {'class': 'search_pagination_right'}).find_all('a')[-2].text)
+        return total_pages
+
+    def _extract_game_info(self, game):
+        """
+        Extracts game information from the BeautifulSoup game element.
+
+        Args:
+            game (BeautifulSoup): The BeautifulSoup element containing game information.
+
+        Returns:
+            tuple: Game information tuple.
+        """
+        name = game.find('span', {'class': 'title'}).text
+        published_date = game.find('div', {'class': 'col search_released responsive_secondrow'}).text.strip()
+        published_date = published_date if published_date else 'N/A'
+
+        div_element = game.find('div', class_='col search_name ellipsis')
+        platform_images = div_element.find_all('span', class_='platform_img')
+        platforms = [img.get('class')[1] if len(img.get('class')) > 1 else None for img in platform_images]
+
+        original_price_elem = game.find('div', {'class': 'discount_original_price'})
+        original_price = original_price_elem.text.strip() if original_price_elem else 'N/A'
+
+        discount_pct_elem = game.find('div', {'class': 'discount_pct'})
+        discount_pct = discount_pct_elem.text.strip() if discount_pct_elem else 'N/A'
+
+        discount_price_elem = game.find('div', {'class': 'discount_final_price'})
+        discount_price = discount_price_elem.text.strip() if discount_price_elem else 'N/A'
+
+        review_summary = game.find('span', {'class': 'search_review_summary'})
+        reviews_html = review_summary['data-tooltip-html'] if review_summary else 'N/A'
+        pattern = r"(.+)<br>(\d+%)\s+of\s+the\s+([\d,]+)\s+user reviews.*"
+        match = re.match(pattern, reviews_html)
+        sentiment = match.group(1) if match else 'N/A'
+        percentage = match.group(2) if match else 'N/A'
+        review_count = match.group(3) if match else 'N/A'
+        review_count = review_count.strip()
+        reviews = f"{sentiment.strip()} - {percentage.strip()}"
+
+        return name, published_date, platforms, original_price, discount_pct, discount_price, reviews, review_count
```

Comment:
Return this as a dictionary with the key names as `name, published_date, platforms, original_price, discount_pct, discount_price, reviews, review_count` and their corresponding values.